kolibri/plugins/device/assets/src/views/FacilitiesPage/index.vue

```diff
@@ -284,6 +287,16 @@
         .catch(() => (this.loadingFacilities = false));
     },
     methods: {
+      isRepeatingTaskCompleted(task, prevTasks) {
+        if (!task.repeat_interval) {
+          return false;
+        }
+        const prevTask = prevTasks.find(({ id }) => id === task.id);
+        if (!prevTask) {
```

Comment:
Hrm, wondering why there is this "previous task" at all, the API should only be returning a single job status from the backend for an id.

If it is happening, liable to be a frontend state persisting which would go away if we refreshed the page.

Reply:
Probably, the name should be improved. Here, by previousTasks, I mean the previous state of the tasks lists. The thing is that for repeating tasks, we never had the task with a "COMPLETED" status, and thereby [this line](https://github.com/learningequality/kolibri/pull/12158/files#diff-000e3e069f5a647e12d6fa6acfec612deb07b16fcc9216515f48066de0cebd84R264) is not executed, and we do not have the updated value of `facility.last_successful_sync` .

The problem is that repeating tasks keep the same reference for multiple executions, and they never end in a "COMPLETED" status because, after the "RUNNING" state, they pass to "QUEUED" again. So the way I found to determine if a single task run has been completed was comparing the current state of the task with the previous snapshot.

Reply:
Yeah - my point was that if this is working it's because we had data from a previous fetch where the task was in progress, so it's quite possible that on a refresh of the page we would still get an issue here.

I don't think this needs to be resolved now - but there's definitely some cleanup we can do overall for how we handle tasks on the frontend (probably through a core useTasks composable that consolidates a lot of this logic).